crates/daphne-server/src/router/extractor.rs

```diff
@@ -159,14 +159,34 @@ where
     }
 }
 
+pub mod dap_sender {
+    pub type DapSender = u8;
+
+    pub const FROM_CLIENT: DapSender = 0;
+    pub const FROM_COLLECTOR: DapSender = 1 << 1;
+    pub const FROM_HELPER: DapSender = 1 << 2;
+    pub const FROM_LEADER: DapSender = 1 << 3;
+
+    pub const fn to_enum(id: DapSender) -> daphne::DapSender {
+        match id {
+            FROM_CLIENT => daphne::DapSender::Client,
+            FROM_COLLECTOR => daphne::DapSender::Collector,
+            FROM_HELPER => daphne::DapSender::Helper,
+            FROM_LEADER => daphne::DapSender::Leader,
+            _ => panic!("invalid dap sender. Please specify a valid dap_sender from the crate::extractor::dap_sender module"),
+        }
+    }
+}
```

Comment:
I guess the purpose of this is to be able to express the sender as a trait bound? That's pretty cool.

Why not just replace `daphne::DapSender` with thee `DapSender` in this module? It seems to me that long term this layer of indirection will get confusing.

Reply:
The purpose of this is to be able to pass the sender as a const generic parameter, which currently only supports very primitive types (numbers). 

Using integers for "enums" has the problem that whenever you match on then you have to include a `_ =>` branch to handle the default value where you probably have to just panic (like I did here), so all the code that currently matches on DapSender will become worse.

This set of constants will never be used outside of these extractors. At least I don't intend it to be used outside of here. And if this is true then I can actually guarantee that this panic never happen at runtime because `to_enum` is `const` and it's [called in a const block](https://github.com/cloudflare/daphne/pull/694/files#diff-a55f2d195924b3d5fef4513679389e84a00d89a0ca8c4b13a26b714be3b9b23cR281), which forces the compiler to execute it at compile time.

Give it a go. Checkout this code and change `FROM_HELPER` [https://github.com/cloudflare/daphne/pull/694/files#diff-963e76984effba09b673f5f9da56571fdb5325e1907af4a2dfd197165d1ead8eR72](here) to `42` and you'll see a compile time error (if you do `cargo build`, sadly `cargo check` doesn't catch it)

Reply:
I see. This makes me sad, but I think it really improves readability here.

Reply:
I'm cool with keeping it.

Another option might be to replace the enum with "marker traits" that are supertraits of `DapSender`. Something to noodle on later.